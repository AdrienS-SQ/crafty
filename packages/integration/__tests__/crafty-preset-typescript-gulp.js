/* global describe, it, expect */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");
const configuration = require("@swissquote/crafty/src/configuration");

const testUtils = require("../utils");

const getCrafty = configuration.getCrafty;

it("Loads crafty-preset-typescript and does not register gulp tasks", () => {
  const crafty = getCrafty(["@swissquote/crafty-preset-typescript"], {});

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-typescript");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([]);
});

it("Loads crafty-preset-typescript, crafty-runner-gulp and registers gulp task", () => {
  const config = { js: { myBundle: { source: "js/**/*.ts" } } };
  const crafty = getCrafty(
    ["@swissquote/crafty-preset-typescript", "@swissquote/crafty-runner-gulp"],
    config
  );

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-typescript");
  expect(loadedPresets).toContain("@swissquote/crafty-runner-gulp");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([
    "js_myBundle",
    "js",
    "default"
  ]);
});

it("Compiles TypeScript", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-gulp/compiles")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();

  expect(fs.existsSync("dist/js/script.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/script.js.map")).toBeTruthy();

  expect(fs.existsSync("dist/js/Component.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/Component.js.map")).toBeTruthy();

  expect(testUtils.readForSnapshot("dist/js/script.js")).toMatchSnapshot();
  expect(testUtils.readForSnapshot("dist/js/Component.js")).toMatchSnapshot();
});

it("Compiles TypeScript and concatenates", () => {
  process.chdir(
    path.join(
      __dirname,
      "../fixtures/crafty-preset-typescript-gulp/concatenates"
    )
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeTruthy();

  expect(fs.existsSync("dist/js/script.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/script.js.map")).toBeFalsy();

  expect(fs.existsSync("dist/js/otherfile.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/otherfile.js.map")).toBeFalsy();

  expect(
    testUtils.readForSnapshot("dist/js/myBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-gulp/fails")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-gulp/lints")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});
