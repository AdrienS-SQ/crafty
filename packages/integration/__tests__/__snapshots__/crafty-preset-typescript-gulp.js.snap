// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles TypeScript 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles TypeScript 2`] = `
"import { test } from \\"./Component\\";
var NewStuff = /** @class */ (function () {
    function NewStuff() {
        // tslint:disable-next-line no-console
        console.log(\\"init class\\");
    }
    NewStuff.prototype.method = function () {
        // tslint:disable-next-line no-console
        console.log(test(2, 4));
        import(\\"./SomeLibrary\\").then(function (SomeLibrary) {
            var som = new SomeLibrary.default();
            // tslint:disable-next-line no-console
            console.log(som.add(2, 4));
        });
    };
    return NewStuff;
}());
export default NewStuff;

//# sourceMappingURL=script.js.map
"
`;

exports[`Compiles TypeScript 3`] = `
"export function test(a, b) {
    return a + b;
}

//# sourceMappingURL=Component.js.map
"
`;

exports[`Compiles TypeScript and concatenates 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/concatenates/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles TypeScript and concatenates 2`] = `
"export function test(a, b) {
    return a + b;
}

var SomeLibrary = /** @class */ (function () {
    function SomeLibrary() {
    }
    SomeLibrary.prototype.add = function (a, b) {
        return a + b;
    };
    SomeLibrary.prototype.substract = function (a, b) {
        return a - b;
    };
    SomeLibrary.prototype.divide = function (a, b) {
        return a / b;
    };
    return SomeLibrary;
}());
export default SomeLibrary;

import { test } from \\"./Component\\";
var NewStuff = /** @class */ (function () {
    function NewStuff() {
        // tslint:disable-next-line no-console
        console.log(\\"init class\\");
    }
    NewStuff.prototype.method = function () {
        // tslint:disable-next-line no-console
        console.log(test(2, 4));
        import(\\"./SomeLibrary\\").then(function (SomeLibrary) {
            var som = new SomeLibrary.default();
            // tslint:disable-next-line no-console
            console.log(som.add(2, 4));
        });
    };
    return NewStuff;
}());
export default NewStuff;

//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 0,
  "stderr": "js/SomeLibrary.ts(10,1): error TS1005: '}' expected.
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] TypeScript: 1 syntax error
[__:__:__] TypeScript: emit succeeded (with errors)
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Lints TypeScript 1`] = `
Object {
  "status": 1,
  "stderr": "[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error in plugin 'gulp-tslint'
Message:
    Failed to lint: __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/Component.ts [2, 24]: missing whitespace, __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/Component.ts [3, 18]: Multiple spaces found before 'b'., __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/SomeLibrary.ts [12, 5]: misplaced opening brace, __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts [13, 5]: misplaced opening brace, __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts [6, 5]: Calls to 'console.log' are not allowed., __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts [10, 5]: Calls to 'console.log' are not allowed., __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts [16, 7]: Calls to 'console.log' are not allowed., __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts [17, 7]: Missing semicolon.
Details:
    domain: [object Object]
    domainThrown: true
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
__PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/Component.ts:2:24
ERROR: 2:24  whitespace       missing whitespace
ERROR: 3:18  no-multi-spaces  Multiple spaces found before 'b'.

__PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/SomeLibrary.ts:12:5
ERROR: 12:5  one-line  misplaced opening brace

__PATH__/packages/integration/fixtures/crafty-preset-typescript-gulp/lints/js/script.ts:6:5
ERROR: 6:5   no-console  Calls to 'console.log' are not allowed.
ERROR: 10:5  no-console  Calls to 'console.log' are not allowed.
ERROR: 13:5  one-line    misplaced opening brace
ERROR: 16:7  no-console  Calls to 'console.log' are not allowed.
ERROR: 17:7  semicolon   Missing semicolon

",
}
`;
